server/routes/add-steps/AddStepsController.ts

```diff
@@ -59,8 +59,7 @@ export default class AddStepsController {
     try {
       await req.services.stepService.saveAllSteps(goalData, req.params.uuid)
 
-      // TODO: this is wrong... we should be able to add steps only, or add steps after adding a (current or future) goal, and get a success banner...
-      const link = req.services.sessionService.getReturnLink() ?? `${URLs.PLAN_OVERVIEW}?status=success`
+      const link = `${URLs.PLAN_OVERVIEW}?type=current`
       req.services.sessionService.setReturnLink(null)
```

Comment:
Same concern here as https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/340/files#r1916598230

Reply:
Yep, no idea what I was thinking with adding these in. I have changed it so that if the returnLink is pointing to change goal, then we will redirect to the plan overview (as it means they are adding a new goal) otherwise we use the return link, this means that if they are changing any steps, such as when it is an agreed plan it will go back to the update goals and steps page.